editor/src/components/canvas/grid-panels-state.tsx

```diff
@@ -63,6 +71,13 @@ export function useGridPanelState() {
     void saveUserPreferencesProjectLayout(projectId, state)
   }, [loaded, state, projectId])
 
+  const permissions = usePermissions()
+  React.useEffect(() => {
+    if (!permissions.edit) {
+      setState((s) => filterCodeEditorFromPanelState(s))
```

Comment:
won't this mean the code editor is re-openable manually?

Reply:
No, the closed panels UI is already using this hook as the source of truth